modules/snapshot/main.tf

```diff
@@ -60,3 +60,14 @@ resource "aws_lambda_permission" "this" {
   function_name       = trimprefix(data.aws_arn.function.resource, "function:")
   source_arn          = aws_cloudwatch_event_rule.trigger.arn
 }
+
+resource "aws_lambda_invocation" "snapshot" {
+  // initial snapshot invocation
+  function_name = data.aws_arn.function.arn
+
+    input = jsonencode({
```

Comment:
Extract this encoded JSON to a local and refer to `local.lambda_snapshot_input` to avoid repeating this in both `aws_cloudwatch_event_target` and this. Or optionally set this to `aws_cloudwatch_event_target.target.input` since it's fine if this is only created after the event target and directly refers to it.

Reply:
👍